messaging/WFLY-13797_galleon_layer_for_remote_artemis.adoc

```diff
@@ -0,0 +1,125 @@
+** The value of the 'port' attribute must be an expression, '${jboss.messaging.connector.port:61616}', to allow externalization of its value. The default of 61616 is the default port used by an ActiveMQ Artemis server.
+* The Galleon layer must NOT provide a server resource in the messaging-activemq subsystem.
+* The Galleon layer must depend on the 'resource-adapters' layer in order to ensure that the necessary resource adapter support is provided.
+* The Galleon layer must configure the 'jms-connection-factory' attribute of the '/subsystem=ee/service=default-bindings' resource to use the 'java:jboss/DefaultJMSConnectionFactory' JNDI binding provided by the pooled-connection-factory=activemq-ra resource.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+* The remote-activemq layer does not expose a management interface nor does it accept incoming connections. That means it doesn't need to depend on any security subsystems.
+* The messaging-activemq subsystem depends on modules that provide libraries needed to run ActiveMQ Artemis as an embedded broker. These same libraries are used to handle connecting to a remote broker. It is not a requirement of this issue to eliminate dependencies on libraries unrelated to the remote-broker use case, although it is acceptable to do so if possible.
+
+== Test Plan
+
+The test coverage of the Galleon layer added by this proposal is divided in three main groups:
+
+. Testing the Galleon layer provisioning. This testing is done by https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]. The testsuite will be modified to add a new server provisioned with this layer in isolation and with this layer combined with all the layers. For each kind of provisioning, this test does the following:
+
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
+
+. Execution of existing WildFly tests related to interaction with a remote broker. Reuse the existing tests available in the WildFly test suite, which are directly testing this layer's functionalities, and execute them on a server installation provisioned with this layer.
+
+. Execution of existing WildFly tests related to interaction with a messaging broker, where the purpose of the test is not to test embedded broker functionality. Most such tests do assume an embedded broker, as WildFly's standard configuration historical includes one. For these we will reuse the existing tests available in the WildFly test suite, but adapted to run with a remote broker configuration, and execute them on a server installation provisioned with this layer.
+
+Note that some of the coverage in the 3rd category of tests may will be delivered as part of this feature but will require othger features,e.g. Galleon layers for EJB (WFLY-13354). Tests of EJB interaction with a broker require both an EJB layer and a broker layer.
```

Comment:
@ehsavoie [TP from EAP7-1524](https://gitlab.mw.lab.eng.bos.redhat.com/jbossqe-eap/test-plans/merge_requests/383/diffs#f4cb0fd3c30feec8140ce1e437f015971caf0a37_0_110) suggests these tests belong here;

Recently there was this same issue with the JSF layer and a combination of JSF+EJB tests was added with  surefire execution [jsf-ejb-lite-layers.surefire in pom.xml](https://github.com/wildfly/wildfly/blob/d21a7f863eb295441457d6b50fc1cf7ff4ec4f20/testsuite/integration/basic/pom.xml) inside [PR#13532](https://github.com/wildfly/wildfly/pull/13532);

AFAIK, use case JMS+EJB is important in the scope of the Bootable jar context: if you agree that's correct, why don't we add some JMS+EJB tests in a similar way as it was achieved for the JSF layer?